src/Qir/Execution/Tools/Executable/QirFullStateExecutable.cs

```diff
@@ -13,20 +16,40 @@ namespace Microsoft.Quantum.Qir.Tools.Executable
     /// </summary>
     public class QirFullStateExecutable : QirExecutable
     {
+        public override string DriverFileExtension => "cpp";
+
+        public override IList<string> LinkLibraries => new List<string> {
+                "Microsoft.Quantum.Qir.Runtime",
+                "Microsoft.Quantum.Qir.QSharp.Foundation",
+                "Microsoft.Quantum.Qir.QSharp.Core"
+            };
+
+        public override IList<DirectoryInfo> HeaderDirectories { get; } = new List<DirectoryInfo>();
+
+        public override IList<DirectoryInfo> LibraryDirectories { get; } = new List<DirectoryInfo>();
+
         public QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger = null)
             : base(executableFile,
                   qirBytecode,
                   new QirFullStateDriverGenerator(),
                   logger)
         {
-        }
+            var thisModulePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            if (string.IsNullOrWhiteSpace(thisModulePath))
+            {
+                throw new InvalidOperationException("Could not get a path for the current assembly location.");
+            }
+            HeaderDirectories.Add(new DirectoryInfo(Path.Combine(thisModulePath, "runtimes", "any", "native", "include")));
+            HeaderDirectories.Add(new DirectoryInfo(Path.Combine(thisModulePath, "Externals", "CLI11")));
 
-        public override IList<string> LinkLibraries => new List<string> {
-                "Microsoft.Quantum.Qir.Runtime",
-                "Microsoft.Quantum.Qir.QSharp.Foundation",
-                "Microsoft.Quantum.Qir.QSharp.Core"
-            };
+            var osID = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "win-x64"
+                : RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? "linux-x64"
+                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "osx-x64"
+                : throw new ArgumentException("Unsupported operating system architecture.");
 
-        public override string DriverFileExtension => "cpp";
+            LibraryDirectories.Add(new DirectoryInfo(Path.Combine(thisModulePath, "runtimes", osID, "native")));
+            LibraryDirectories.Add(new DirectoryInfo(Path.Combine(thisModulePath, "Libraries", osID)));
+            LibraryDirectories.Add(new DirectoryInfo(Path.Combine(thisModulePath, "Libraries")));
```

Comment:
Since library directories are copied into the bin folder, won't this have the effect of copying the contents of `Libraries/<osId>` twice, once directly into the bin folder and once as subdir of the bin folder?

Reply:
These are the directories that get passed to the clang compiler. I don't believe it checks for files recursively, and I don't think we do on our end either, so the reference to the Libraries directory is not sufficient to get the files in the subdirectory by itself. The simulator library file is directly under the Libraries directory, while other platform-specific libraries are further nested. We tried adding the simulator library to the platform-specific subdirectories, but that had other side-effects related to how MSBuild expects the files to be organized. Listing both directories here should not cause duplicate copies to be copied to the output folder of a Q# program compiled with the tool.

Reply:
Oh, I see! The subdirectory is not copied by `CopyDirectoryContents` because that method doesn't include folders. That makes sense, thanks for explaining!